src/hidimstat/base_perturbation.py

```diff
@@ -156,7 +156,14 @@ def importance(self, X, y):
 
     def _check_fit(self):
         """Check that the estimator has been fitted if needed."""
-        pass
+        if (
+            self.n_groups is None
+            or not hasattr(self, "groups")
+            or not hasattr(self, "_groups_ids")
```

Comment:
Just wondering: how do "groups" and  "_groups_ids" differ ? in which situation would we have one and not the other ?

Reply:
When X is a pandas DataFrame, we allow groups to be passed as column names. In that case, `_groups_ids` would be the column indices (int) when considering the corresponding array and `groups` the column names (str).